src/Compilers/Core/Portable/Syntax/SyntaxList.SeparatedWithManyChildren.cs

```diff
@@ -40,21 +40,28 @@ internal SeparatedWithManyChildren(InternalSyntax.SyntaxList green, SyntaxNode?
 
             internal override int GetChildPosition(int index)
             {
-                // If the previous sibling (ignoring separator) is not cached, but the next sibling
-                // (ignoring separator) is cached, use the next sibling to determine position.
-                int valueIndex = (index & 1) != 0 ? index - 1 : index;
-                // The check for valueIndex >= Green.SlotCount - 2 ignores the last item because the last item
-                // is a separator and separators are not cached. In those cases, when the index represents
-                // the last or next to last item, we still want to calculate the position from the end of
-                // the list rather than the start.
-                if (valueIndex > 1
-                    && GetCachedSlot(valueIndex - 2) is null
-                    && (valueIndex >= Green.SlotCount - 2 || GetCachedSlot(valueIndex + 2) is { }))
+                int leftNodeIndex = (index >> 1) - 1;
+                var children = _children;
+
+                // If the closest node on the left is uncached
+                if (unchecked((uint)leftNodeIndex < (uint)children.Length) && children[leftNodeIndex].Value is null)
```

Comment:
same issue iwth the unchecked code.  please undo in all the locations, or explain what it is for.

Reply:
see comment above about the two checks for the price of one that this common pattern enables